### PR TITLE
Modify RealmAdminResource.partialImport to work with InputStream

### LINKED ISSUE #13611: Modify `RealmAdminResource.partialImport` to work with `InputStream`
Context: see #13609

This task is to prepare API so that the endpoint consumes `InputStream` and uses legacy / new store implementation logic to process the stream accordingly. This implies relevant changes in `PartialImportManager`, pushing the conversion from the stream to `PartialImportRepresentation` as much to the storage layer as possible.

### PR DESCRIPTION
**Pull request by ahus1**
Rework existing PartialImportManager to not interfere with transaction handling, and bundle everything related to AdminEventBuild and JAX-RS Repsonses inside the Resource.

Closes #13611

### CODE REVIEW

**Comment by ahus1** [comment]
This worked for a local partial import with both the legacy and the map storage, now waiting for a first set of results from the GHAs.

Ideas that went into this: 

* `PartialImportManager` is tightly coupled to the logical layer as it interacts with it quite a bit
* The different `PartialImport` implementations rely on the fact that the representation is created once, as they write the new IDs to the representation. Therefore the `PartialImportManager` should be called with one already de-serialized representation.
* To allow accessing the existing `PartialImport` logic from the `MapExportImportManager`, it is using again an event to "bubble upwards"
* The transaction handling was manual within the `PartialImport`, and the exceptions where mapped to a `Response` inside the code. As the `ExportImportManager` shouldn't interact with the transaction, and should not depend on the `Reponse`, the logic has been moved to the `Resource`.
* The `PartialImportResults` and the classes it depend on have been moved from `keycloak-services` to `server-spi-private` to allow this being a return time of `ExportImportManager`.

**Review by hmlnarik — COMMENTED** [review]
Thank you for the PR, @ahus1!

I like this PR as it represents a right step to the ultimate solution which will remove the need of keeping legacy representations in core keycloak module and would allow moving it to more appropriate one.

File: server-spi-private/src/main/java/org/keycloak/models/utils/KeycloakModelUtils.java [hunk 1]
```diff
@@ -276,6 +276,35 @@ public static void runJobInTransaction(KeycloakSessionFactory factory, KeycloakS
         }
     }
 
+    /**
+     * Wrap a given callable job into a KeycloakTransaction.
+     */
+    public static <V> V runJobInTransactionWithResult(KeycloakSessionFactory factory, final KeycloakSessionTaskWithResult<V> callable) {
```
[the remark this answers is not in the document]
Comment (ahus1) [reply server-spi-private/src/main/java/org/keycloak/models/utils/KeycloakModelUtils.java:262]: sure, force-pushed the changes.

File: server-spi-private/src/main/java/org/keycloak/storage/ExportImportManager.java [hunk 2]
```diff
@@ -34,6 +35,8 @@
 public interface ExportImportManager {
     void importRealm(RealmRepresentation rep, RealmModel newRealm, boolean skipUserDependent);
 
+    PartialImportResults partialImportRealm(InputStream requestBody, RealmModel realm);
```
[the remark this answers is not in the document]
Comment (ahus1) [reply server-spi-private/src/main/java/org/keycloak/storage/ExportImportManager.java]: sure, force-pushed the changes.

**Comment by ahus1** [comment]
Previous tests were green, let's hope they will be green again. Switching this to "ready".

**Review by hmlnarik — APPROVED** [review]
Thank you for the PR!

**Pull request merged**